File: algorithms/2073.time-needed-to-buy-tickets/solution.rs
impl Solution {
    pub fn time_required_to_buy(tickets: Vec<i32>, k: i32) -> i32 {
        let mut seconds = 0i32;
        let mut desires = tickets.clone();

        loop {
            for i in 0..desires.len() {
                if desires[i] > 0 {
                    seconds += 1;
                    desires[i] -= 1;

                    if k as usize == i && desires[i] == 0 {
                        return seconds;
                    }
                }
            }
        }

        seconds
    }
}

File: algorithms/2073.time-needed-to-buy-tickets/solution.cpp
#include <vector>

using namespace std;

class Solution {
public:
    int timeRequiredToBuy(vector<int>& tickets, int k) {
        int seconds = 0;
        int n = tickets.size();

        while (true) {
            for (int i = 0; i < n; i++) {
                if (tickets[i] > 0) {
                    seconds++;
                    tickets[i]--;

                    if (k == i && tickets[i] == 0) {
                        return seconds;
                    }
                }
            }
        }

        return seconds;
    }
};
